clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/CompletableApplicationEvent.java

```diff
@@ -36,13 +36,14 @@ protected CompletableApplicationEvent(final Type type, final Timer timer) {
         super(type);
         this.future = new CompletableFuture<>();
         Objects.requireNonNull(timer);
-        this.deadlineMs = timer.remainingMs() + timer.currentTimeMs();
-    }
 
-    protected CompletableApplicationEvent(final Type type, final long deadlineMs) {
-        super(type);
-        this.future = new CompletableFuture<>();
-        this.deadlineMs = deadlineMs;
+        long currentTimeMs = timer.currentTimeMs();
+        long remainingMs = timer.remainingMs();
+
+        if (currentTimeMs > Long.MAX_VALUE - remainingMs)
+            this.deadlineMs = Long.MAX_VALUE;
+        else
+            this.deadlineMs = currentTimeMs + remainingMs;
```

Comment:
`CompleteableEvent` is an interface, but I could see if I can put a static method in there to keep the logic in one place.

Reply:
For point #1, I created a new method named `calculateDeadlineMs` that moves this code into one place.

For point #2, there is no `Timer` in the event because `Timer` is not thread safe, and events cross the application/background thread boundary. I did not want to expose the `Timer` in the event to avoid its possible usage from the background thread.